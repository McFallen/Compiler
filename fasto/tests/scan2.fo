fun int multi(int a, int b) = a * b

fun int writeInt(int n) = write(n)

fun [int] main() =
  let a = {1, 2, 3, 4} in
  let b = map(fn int (int x) => x*2, a) in 
  let c = scan(multi, 1, b) in
  map(writeInt, c)