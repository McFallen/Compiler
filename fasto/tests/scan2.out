1 2 8 48 384